src/main/groovy/org/shipkit/gradle/UpdateReleaseNotesTask.java

```diff
@@ -41,11 +44,34 @@
     private Collection<String> contributors;
     private File contributorsDataFile;
     private boolean emphasizeVersion;
+    private IncrementalNotesGenerator incrementalNotesGenerator = new IncrementalNotesGenerator();
+
+    /**
+     * Generates incremental release notes and appends it to the top of release notes file.
+     * Run with -Ppreview if you only want to see preview of generated release notes, without appending them to the file.
+     */
+    @TaskAction
+    public void updateReleaseNotes() {
+        String newContent = getNewContent();
+        if (isInPreviewMode()){
+            LOG.lifecycle("  Preview of release notes update:\n" +
+                    "  ----------------\n" + newContent + "----------------");
+        } else{
+            FileUtil.appendToTop(newContent, getReleaseNotesFile());
+            LOG.lifecycle("  Successfully updated release notes!");
+        }
+    }
+
+    /**
+     * @return true if task is configured to generate only preview of release notes, and false otherwise
+     */
+    public boolean isInPreviewMode() {
```

Comment:
Instead of inspecting the project property on the fly I suggest to create a boolean property on the task "preview" or "previewMode". This is more natural in Gradle and also makes the task more flexible for folks who want to use the task but not necessarily configure it via project property. The plugin would configure the boolean property based on the project property.